src/arch/ucx/machine.C

```diff
@@ -614,8 +619,11 @@ void LrtsAdvanceCommunication(int whileidle)
     do {
        cnt = ucp_worker_progress(ucxCtx.worker);
 
+       // Probe with full tag mask to avoid long traversing thru unexpected
+       // queue of eager messages (messages with non-full mask added to the
+       // same unexpected queue)
        msg = ucp_tag_probe_nb(ucxCtx.worker, UCX_MSG_TAG_PROBE,
-                              UCX_MSG_TAG_MASK, 1, &info);
+                              UCX_MSG_TAG_MASK_FULL, 1, &info);
```

Comment:
The overall change looks good. 

I have one question: Before this change, which messages were not being caught by this probe call? (i.e. messages with what tags were having to go through the unexpected queue?). I'm asking this because I see that the earlier tag `UCX_MSG_TAG_MASK` (00001111) was catching all messages except the UCX_RMA tags right?

Reply:
@nitbhat, previously probe was done with non-full mask matching the messages sent with `PROBE` bit (0x2).  This bit is used for sending long messages (regardless whether it is auxiliary zcopy message or payload). The problem is with probing using non-full tag mask. In this case UCX will go thru the list of all unexpected messages before it finds the match (or does not find if none such messages). With Changa there were a lot of small messages sent with eager protocol (i.e. which are not mathcing the probe for long messages). But with every LrtsAdvance call UCX tried to find long message going thru all the list of small messages. 
Now, when we do probe for the full tag, UCX search for the message just in a single queue intended specifically for this  `PROBE` tag. And if there is no any long message arrived probe function will return immediately.

Also to probe with full mask I have to restrict auxilliary messages to be sent with eager protocol only, because I use some high bits of the tag marking the type of the message.

Reply:
Oh okay, I understand. Thanks for explaining that. It will be useful to add some of this information in the commit message. Something like:

`This commit changes the ucp_tag_probe_nb call to probe with a full tag mask instead of a non-full tag mask. This is done in order to search for PROBE tag messages (used for large messages) in a separate queue and avoid iterating over the small message queues for every LrtsAdvanceCommunication call, hence optimizing receives, especially for applications with a large number of short messages.`